fdbcli/DebugCommands.actor.cpp

```diff
@@ -164,125 +164,133 @@ ACTOR Future<bool> getallCommandActor(Database cx, std::vector<StringRef> tokens
+	// Decide comparison scope
+	Key claimEndKey; // used for the next round exists
+	bool hasMore = false;
 	for (int j = 0; j < replies.size(); j++) {
 		auto reply = replies[j].get();
-		if (!reply.present() || reply.get().error.present()) {
-			printf("Error from %s: %s\n",
-			       servers[j].address().toString().c_str(),
-			       reply.present() ? reply.get().error.get().what() : "no reply");
-			continue;
+		if (reply.isError()) {
+			printf("checkResults error: %s\n", reply.getError().what());
+			throw reply.getError();
+		} else if (reply.get().error.present()) {
+			printf("checkResults error: %s\n", reply.get().error.get().what());
+			throw reply.get().error.get();
 		}
 		GetKeyValuesReply current = reply.get();
+		if (current.data.size() == 0) {
+			continue; // Ignore if no data has replied
+		}
+		if (claimEndKey.empty() || current.data[current.data.size() - 1].key < claimEndKey) {
+			claimEndKey = current.data[current.data.size() - 1].key;
+		}
+		hasMore = hasMore || current.more;
+	}
+	ASSERT(!claimEndKey.empty());
+
+	// Compare servers
+	bool allSame = true;
+	int firstValidServer = -1;
+	for (int j = 0; j < replies.size(); j++) {
+		auto reply = replies[j].get();
+		ASSERT(reply.present() && !reply.get().error.present()); // has thrown eariler of error
 		if (firstValidServer == -1) {
 			firstValidServer = j;
-			continue;
+			continue; // always select the first server as reference
 		}
+		// compare reference and current
+		GetKeyValuesReply current = reply.get();
 		GetKeyValuesReply reference = replies[firstValidServer].get().get();
-
-		// compare the two replicas
 		if (current.data == reference.data && current.more == reference.more) {
 			continue;
 		}
 
+		// Detecting corrupted keys for any mismatching replies between current and reference servers
 		allSame = false;
-		printf("#%d  server: %s  key count: %lu, cached: %d, more: %d\n",
-		       firstValidServer,
-		       servers[firstValidServer].address().toString().c_str(),
-		       reference.data.size(),
-		       reference.cached,
-		       +reference.more);
-		printf("#%d  server: %s  key count: %lu, cached: %d, more: %dn",
-		       j,
-		       servers[j].address().toString().c_str(),
-		       current.data.size(),
-		       current.cached,
-		       current.more);
 		size_t currentI = 0, referenceI = 0;
 		while (currentI < current.data.size() || referenceI < reference.data.size()) {
+			if (hasMore &&
+			    (reference.data[referenceI].key >= claimEndKey || current.data[currentI].key >= claimEndKey)) {
+				// If there will be a next round and the key is out of claimEndKey
+				// We will delay the detection to the next round
+				break;
+			}
 			if (currentI >= current.data.size()) {
-				printf(" #%d CurrentI: %lu ReferenceI: %lu Unique key: %s\n",
-				       firstValidServer,
+				printf("UniqueKey, %s(1), %s(0), CurrentIndex %lu, ReferenceIndex %lu, Version %ld, UniqueKey %s\n",
```

Comment:
what's the meaning of 1 and 0 here, i.e., "%s(1), %s(0)"?
`UniqueKey` appears twice.

Reply:
1 means that the key appears in the server right before "1". 0 means that the key does not exist in the server right before "0".